Pop-Grupa-A-Frontend/src/components/computationCockpit/applicationList/AppList.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+
+class AppList extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      chosenAppId: -1,
+      userId: 1,
+    };
+    this.createTask = this.createTask.bind(this);
+  }
+
+  onAppClick = (chosenAppId) => {
+    this.setState({ chosenAppId });
+  }
+
+  createTask(e) {
+    e.preventDefault();
+    fetch("http://localhost:5000/task/", {
+      crossDomain:  true,
+      method: 'post',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        app_id: this.state.chosenAppId,
+        user_id: this.state.userId
+      })
+    })
+    .then(response => response.json())
+    .then(response => {
+      this.setState({
+        chosenAppId: -1
+      })
+      alert(response.message);
+    })
+    this.props.updateTaskList(true)
```

Comment:
wystarczy,że ktoś raz stworzy taska i flaga `newTaskCreated` zawszę będzie prawdziwa.
Po odświeżeniu listy tasków w TaskList.js trzeba ustawić flagę newTaskCreated(true). Dzięki temu kontrolujecie kiedy odświeżana jest lista tasków.
Jak widzę, chcecie w Tasklist.js zrobić sprawdzenie czy został stworzony nowy Task i dopiero potem wywołać te metodę?
To w metodzie ComponentDidUpdate trzeba porównać starą i nową wartość newTaskCreated plus sprawdzić czy ona zmieniła się z false na true.

Reply:
Up

Reply:
Poprawione chyba tak jak chciałeś